packages/core/src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -741,4 +745,82 @@ export class AbstractQueryInterfaceTypeScript<Dialect extends AbstractDialect =
 
     return this.sequelize.queryRaw(sql, { ...bulkDeleteOptions, raw: true, type: QueryTypes.DELETE });
   }
+
+  /**
+   * Update multiple records of a table
+   *
+   * @example
+   * queryInterface.bulkUpdate('roles', {
+   *     label: 'admin',
+   *   }, {
+   *     userType: 3,
+   *   },
+   * );
+   *
+   * @param tableName         Table name to update
+   * @param values            Values to be inserted, mapped to field name
+   * @param options           Various options, please see Model.bulkCreate options
+   * @param columnDefinitions Attributes on return objects if supported by SQL dialect
+   */
+  async bulkUpdate(
+    tableName: TableNameOrModel,
+    values: Record<string, unknown>,
+    options?: QiUpdateOptions,
+    columnDefinitions?: Record<string, AttributeOptions>,
+  ): Promise<number> {
+    if (options?.bind) {
+      assertNoReservedBind(options.bind);
+    }
+
+    const bulkUpdateOptions = { ...options };
+    const { bind, query } = this.queryGenerator.updateQuery(tableName, values, bulkUpdateOptions, columnDefinitions);
+
+    // unlike bind, replacements are handled by QueryGenerator, not QueryRaw
+    delete bulkUpdateOptions.replacements;
+    bulkUpdateOptions.bind = combineBinds(bulkUpdateOptions.bind ?? {}, bind ?? {});
```

Comment:
to reduce the number of objects we create, instead of defaulting to `{}`, we could make `combineBinds` accept `undefined` for both parameters and return either if the other is undefined

Reply:
See comment below